File: sorting/Leetcode-973-KClosestPointsToOrigin/Leetcode-973-KClosestPointsToOrigin.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        sort(points.begin(), points.end(), [](auto a, auto b){
            return pow(a[0],2)+pow(a[1],2) < pow(b[0],2)+pow(b[1],2);
        });

        vector<vector<int>> result(points.begin(), points.begin() + k);

        return result;
    }
};

File: maureen72/Sort/Leetcode-973-K_ClosestPointstoOrigin.py
class Solution(object):
    def kClosest(self, points, k):
        """
        :type points: List[List[int]]
        :type k: int
        :rtype: List[List[int]]
        """
        lst = []
        for (x,y) in points:
            distance = x**2 + y**2
            lst.append((distance, x, y))
        lst.sort()
        return [[x, y] for (_, x, y) in lst[:k]]

        